kong/plugins/key-auth/handler.lua

```diff
@@ -17,13 +17,20 @@ local KeyAuthHandler = BasePlugin:extend()
 
 KeyAuthHandler.PRIORITY = 1000
 
+local OPTIONS = "OPTIONS"
+
 function KeyAuthHandler:new()
   KeyAuthHandler.super.new(self, "key-auth")
 end
 
 function KeyAuthHandler:access(conf)
   KeyAuthHandler.super.access(self)
 
+  -- do not authenticate preflight OPTIONS requests
+  if ngx.req.get_method() == OPTIONS then
+    return
+  end
+
```

Comment:
I don't think there is ever a point in authenticating OPTIONS requests... Could be wrong, would love to see a use-case for it.
It would be worth adding a debug log when this plugin receives an OPTIONS request maybe?